scylla-cdc-replicator/src/replicator_consumer.rs

```diff
@@ -167,6 +167,70 @@ impl ReplicatorConsumer {
         Ok(())
     }
 
+    // Recreates INSERT/DELETE/UPDATE statement on non-frozen list.
+    async fn update_list<'a>(
+        &self,
+        column_name: &str,
+        data: &'a CDCRow<'_>,
+        values_for_update: &mut Vec<&'a CqlValue>,
+        values_for_remove: &mut Vec<&'a CqlValue>,
+        values_for_delete: &[&CqlValue],
+        timestamp: i64,
+    ) -> anyhow::Result<()> {
+        if data.is_value_deleted(column_name) {
+            // If the list was replaced with syntax "(...) SET l = [(...)]",
+            // we have to manually delete the list
+            // and add every element with timeuuid from the original table,
+            // because we want to preserve the timestamps of the list elements.
+            // The list is deleted with timestamp decreased by one,
+            // this is how Scylla does it internally.
+            self.run_statement(
+                Query::new(format!(
+                    "DELETE {col} FROM {ks}.{tbl} WHERE {cond}",
+                    col = column_name,
+                    ks = self.dest_keyspace_name,
+                    tbl = self.dest_table_name,
+                    cond = self.keys_cond,
+                )),
+                values_for_delete,
+                timestamp - 1,
+            )
+            .await?;
+        }
+
+        let add_query = Query::new(
+            format!("UPDATE {ks}.{tbl} USING TTL ? SET {list}[SCYLLA_TIMEUUID_LIST_INDEX(?)] = ? WHERE {cond}",
+                             ks = self.dest_keyspace_name,
+                             tbl = self.dest_table_name,
+                             list = column_name,
+                             cond = self.keys_cond));
```

Comment:
The indentation is a bit strange here. Why not indent it as you did e.g. for the first invocation of `self.run_statement` in this function (`update_list`)?